File: demo-service-impl/src/main/java/ru/zubmike/service/demo/dao/PlanetarySystemDao.java
package ru.zubmike.service.demo.dao;

import ru.zubmike.core.dao.DictItemDao;
import ru.zubmike.service.demo.types.PlanetarySystem;

import java.util.Optional;

public interface PlanetarySystemDao extends DictItemDao<Integer, PlanetarySystem> {

	Optional<PlanetarySystem> getByCode(String code);

}

File: demo-service-kotlin-impl/src/main/kotlin/ru/zubmike/service/demo/dao/PlanetarySystemDao.kt
package ru.zubmike.service.demo.dao

import ru.zubmike.core.dao.DictItemDao
import ru.zubmike.service.demo.types.PlanetarySystem

interface PlanetarySystemDao : DictItemDao<Int, PlanetarySystem> {

    fun getByCode(code: String): PlanetarySystem?

}
